File: backend/internal/model/vo/finding.go
package vo

import "antd-gin-admin-backend/internal/model/entity"

// FindingVO is the read-only API view of a Finding.
type FindingVO struct {
	FindingCode     string `json:"finding_code"`
	JobCode         string `json:"job_code"`
	RuleCode        string `json:"rule_code"`
	RuleDisplayName string `json:"rule_display_name"`
	Severity        string `json:"severity"`
	Title           string `json:"title"`
	Description     string `json:"description"`
	Evidence        string `json:"evidence"`
	Location        string `json:"location"`
	CreatedAt       string `json:"created_at"`
}

func BuildFindingVO(f *entity.Finding, ruleDisplayName string) *FindingVO {
	if f == nil {
		return nil
	}
	return &FindingVO{
		FindingCode:     f.FindingCode,
		JobCode:         f.JobCode,
		RuleCode:        f.RuleCode,
		RuleDisplayName: ruleDisplayName,
		Severity:        f.Severity,
		Title:           f.Title,
		Description:     f.Description,
		Evidence:        f.Evidence,
		Location:        f.Location,
		CreatedAt:       f.CreatedAt.Format("2006-01-02 15:04:05"),
	}
}

File: frontend/admin-web/src/services/antd-gin-api/finding.ts
import { request } from '@umijs/max';
import type { PageResult } from './asset';

export interface Finding {
  finding_code: string;
  job_code: string;
  rule_code: string;
  rule_display_name: string;
  severity: 'critical' | 'high' | 'medium' | 'low' | 'info';
  title: string;
  description: string;
  evidence: string;
  location: string;
  created_at: string;
}

export interface ListFindingParams {
  page?: number;
  pageSize?: number;
  job_code?: string;
  rule_code?: string;
  severity?: string;
  keyword?: string;
}

/** 获取发现项列表 */
export async function getFindingList(params?: ListFindingParams) {
  return request<{
    code: number;
    data: PageResult<Finding>;
    message?: string;
  }>('/api/v1/scan/findings', {
    method: 'GET',
    params,
  });
}

/** 获取发现项详情 */
export async function getFinding(findingCode: string) {
  return request<{
    code: number;
    data: Finding;
    message?: string;
  }>(`/api/v1/scan/findings/${findingCode}`, {
    method: 'GET',
  });
}
